nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -38,7 +38,7 @@ object Type {
   final object F64 extends F(64)
 
   final case class Array(ty: Type, n: Int)              extends Type
-  final case class Function(args: Seq[Type], ret: Type) extends Type
+  final case class Function(args: Seq[Arg], ret: Type)  extends Type
```

Comment:
Another thing that would be good to add here is `isExtern` boolean flag to the function type. This will make easy to fix #122 afterwards.